### PR TITLE
Add new tags and JSON-LD support

### PR DESCRIPTION
**Pull request by haydenbleasel**

### CODE REVIEW

File: Meta.svelte [hunk 1]
```diff
@@ -3,15 +3,74 @@
...
-  /** Page description */
+  /** Page description **/
   export let description: Optional<string> = undefined;
-  /** Page image */
+  /** Page image **/
   export let image: Optional<string> = undefined;
+  /** Page url **/
+  export let url: Optional<string> = undefined;
+  /** Site logo **/
+  export let logoUrl: string = url ? `${url}/logo.png` : undefined;
```
Comment (madeleineostoja) [inline Meta.svelte]: This seems too opinionated and magic a default, let's just leave as separate to URL. Also since URL can change per page, it would generate bad links on page change. Unless `url` is meant to be root site URL only, in which case the prop name should change

File: Meta.svelte [hunk 2]
```diff
@@ -3,15 +3,74 @@
...
+  /** Search URL **/
+  export let searchUrl: Optional<string> = undefined;
+  /** Sitemap URL **/
+  export let sitemapUrl: string = url ? `${url}/sitemap.xml` : undefined;
```
Comment (madeleineostoja) [inline Meta.svelte]: See notes on `logoUrl`

File: Meta.svelte [hunk 3]
```diff
@@ -3,15 +3,74 @@
...
+  /** Facebook App ID **/
+  export let facebookAppId: Optional<string> = undefined;
 </script>
 
 <svelte:head>
+
+  <meta charset="utf-8" />
+  <meta http-equiv="x-ua-compatible" content="ie=edge" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+  <meta name="robots" content="index,follow" />
```
Comment (madeleineostoja) [inline Meta.svelte]: Isn't this quite opinionated to hardcode?

File: Meta.svelte [hunk 4]
```diff
@@ -3,15 +3,74 @@
...
+  /** Page url **/
+  export let url: Optional<string> = undefined;
+  /** Site logo **/
+  export let logoUrl: string = url ? `${url}/logo.png` : undefined;
+  /** Search URL **/
+  export let searchUrl: Optional<string> = undefined;
+  /** Sitemap URL **/
+  export let sitemapUrl: string = url ? `${url}/sitemap.xml` : undefined;
+  /** Twitter Options **/
+  export let twitter: any = {
```
Comment (madeleineostoja) [inline Meta.svelte]: Wdyt about instead just giving freeform `key:value` options, since these are all just very light sugar on individual props (unlike title, description, etc)

eg: 

```
<Meta custom={{
 'twitter:card': 'summary_large_image',
 'og:locale': 'en_US'
}} />
```

File: Meta.svelte [hunk 5]
```diff
@@ -3,15 +3,74 @@
...
+  export let openGraph: any = {
+    type: undefined,
+    imageWidth: undefined,
+    imageHeight: undefined,
+    locale: 'en_US',
+    siteName: undefined,
+    imageAlt: title || undefined,
+  }
+  /** Facebook App ID **/
+  export let facebookAppId: Optional<string> = undefined;
```
Comment (madeleineostoja) [inline Meta.svelte]: This seems out of scope for social meta? Isn't this for registered OAuth apps/pixel tracking?

File: Meta.svelte [hunk 6]
```diff
@@ -3,15 +3,74 @@
...
 </script>
 
 <svelte:head>
+
+  <meta charset="utf-8" />
+  <meta http-equiv="x-ua-compatible" content="ie=edge" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+  <meta name="robots" content="index,follow" />
+  <meta name="googlebot" content="index,follow" />
+  <link rel="sitemap" type="application/xml" href="{sitemapUrl}" />
```
Comment (madeleineostoja) [inline Meta.svelte]: Should be conditional to avoid 404s

File: Meta.svelte [hunk 7]
```diff
@@ -3,15 +3,74 @@
...
+
+  <meta name="twitter:card" content="{twitter.card}" />
```
Comment (madeleineostoja) [inline Meta.svelte]: Should be conditional rather than having a default? Otherwise could be setting things you're not aware you're setting

File: Meta.svelte [hunk 8]
```diff
@@ -25,5 +84,51 @@
 
   {#if image}
     <meta property="og:image" content={image} />
```
Comment (madeleineostoja) [inline Meta.svelte]: Let's make `image` an `object | string` prop instead

```html
<!-- Only sets og:image -->
<Meta image="/url-of-image.png" />
```

```html
<!-- Sets all props defined -->
<Meta image={{
  src: '/url-of-image.png`,
  height: 400,
  width: 600,
  alt: 'A cover image'
}} />
```

**Review by madeleineostoja — CHANGES_REQUESTED** [review]
Left some comments, let me know what you think!

**Comment by madeleineostoja** [comment]
On the `url` prop, from looking at the tags it generates should it only be the site root URL, rather than the page's URL? If so let's rename it to `siteUrl` to avoid confusion

File: Meta.svelte (hunk 2 above)
Comment (madeleineostoja) [reply Meta.svelte]: See additional comment below. This would be a good default if `url` is actually `siteUrl` and I misunderstood

File: Meta.svelte (hunk 1 above)
Comment (haydenbleasel) [reply Meta.svelte]: Good call, I overlooked the changing URL. Making this `undefined`.

File: Meta.svelte (hunk 2 above)
Comment (haydenbleasel) [reply Meta.svelte]: URL is page-based (not site-wide), changing `sitemapUrl` to be `undefined`.

File: Meta.svelte (hunk 3 above)
Comment (haydenbleasel) [reply Meta.svelte]: I feel like these tags specifically are standard, I haven't seen a site with modified versions of this tag in the last few years.

File: Meta.svelte (hunk 4 above)
Comment (haydenbleasel) [reply Meta.svelte]: I thought we could write interfaces for OpenGraph and Twitter props to prevent typos.

File: Meta.svelte (hunk 5 above)
Comment (haydenbleasel) [reply Meta.svelte]: Partially, also for Facebook Insights, but yeah happy to leave this out.

File: Meta.svelte (hunk 7 above)
Comment (haydenbleasel) [reply Meta.svelte]: Removing the default for now, but would be good to have enums for these.

File: Meta.svelte (hunk 8 above)
Comment (haydenbleasel) [reply Meta.svelte]: Sounds good :+1:

File: Meta.svelte (hunk 3 above)
Comment (madeleineostoja) [reply Meta.svelte]: Even the robots?? Oh would a robots.txt override it?

File: Meta.svelte (hunk 4 above)
Comment (madeleineostoja) [reply Meta.svelte]: Hmm good call. Maybe we can leave Twitter and OG interfaces for another PR, I think it needs a bit of thinking.

~~Maybe we can still make it more abstracted with some sugar, etc~~

```
twitter={{
 card: '',
 image: ''
}}
openGraph={{
  ...
}}
```

EDIT — just realised this is pretty much the same interface as the original PR, though for implementation I think the sub-props would be arbitrary, mapping them to `twitter:` and `og:` tags. Rather than hard-coding and he available props.

File: Meta.svelte (hunk 7 above)
Comment (madeleineostoja) [reply Meta.svelte]: Yeah see above comments on impl deets for `twitter` prop, if we include it in this PR at all

File: Meta.svelte (hunk 4 above)
Comment (madeleineostoja) [reply Meta.svelte]: If the sub props include all possible og tags/are arbitrary, people could also use it to override the default page title for a particular network for eg

File: Meta.svelte (hunk 3 above)
Comment (haydenbleasel) [reply Meta.svelte]: We could make robots a variable and set the default to `index,follow`? I feel like that's a fair standard.

File: Meta.svelte [hunk 9]
```diff
@@ -3,27 +3,133 @@
...
+  /** Page url **/
+  export let url: Optional<string> = undefined;
+  /** Site logo **/
+  export let logoUrl: Optional<string> = undefined;
+  /** Search URL **/
+  export let searchUrl: Optional<string> = undefined;
+  /** Sitemap URL **/
+  export let sitemapUrl: Optional<string> = undefined;
+  /** Twitter Options **/
+  export let twitter: TwitterCardProps = {
```
Comment (madeleineostoja) [inline Meta.svelte:54]: The defaults don't match the interface, should prob add all possible types to the interface (also it can be a type, since it doesn't need to be extended for etc)

File: Meta.svelte [hunk 10]
```diff
@@ -3,27 +3,133 @@
...
+    title,
+    description,
+    image: image ? image.url : undefined,
+    'image:alt': image ? image.alt : undefined,
+    card: undefined,
```
Comment (madeleineostoja) [inline Meta.svelte]: Don't need to define these, they're covered in the interface/type
